ProcessLib/ThermalTwoPhaseFlowWithPP/ThermalTwoPhaseFlowWithPPLocalAssembler-impl.h

```diff
@@ -0,0 +1,473 @@
+        // molar fraction of water vapor in nonwet phase
+        double const x_vapor_nonwet = p_vapor_nonwet / pg_int_pt;
+
+        // mass fraction of air in the nonwet phase
+        double const X_air_nonwet =
+            x_air_nonwet / (x_air_nonwet + x_vapor_nonwet * Water / Air);
+        double const rho_mol_nonwet = pg_int_pt / IdealGasConstant / T_int_pt;
+        double const rho_mol_water = rho_water / Water;
+
+        double const d_rho_mol_nonwet_d_pg = 1 / IdealGasConstant / T_int_pt;
+        double const d_p_vapor_nonwet_d_T =
+            _process_data.material->calculateDerivativedPgwdT(
+                pc_int_pt, T_int_pt, rho_water);
+        double const d_p_vapor_nonwet_d_pc =
+            _process_data.material->calculateDerivativedPgwdPC(
+                pc_int_pt, T_int_pt, rho_water);
+        double const d_rho_mol_nonwet_d_T =
+            -pg_int_pt / IdealGasConstant / T_int_pt / T_int_pt;
+        double const d_x_air_nonwet_d_pg =
+            p_vapor_nonwet / pg_int_pt / pg_int_pt;
+        double const d_x_air_nonwet_d_pc = -d_p_vapor_nonwet_d_pc / pg_int_pt;
+        double const d_x_air_nonwet_d_T = -d_p_vapor_nonwet_d_T / pg_int_pt;
+
+        double const rho_nonwet_air =
+            p_air_nonwet * Air / IdealGasConstant / T_int_pt;
+        double const rho_nonwet_vapor =
+            p_vapor_nonwet * Water / IdealGasConstant / T_int_pt;
+        double const rho_nonwet = rho_nonwet_air + rho_nonwet_vapor;
+        double const rho_wet = rho_water;
+        double const rho_solid = _process_data.density_solid(t, pos)[0];
+        // Derivative of nonwet phase density in terms of T
+        double const d_rho_nonwet_d_T =
+            _process_data.material->calculatedRhoNonwetdT(
+                p_air_nonwet, p_vapor_nonwet, pc_int_pt, T_int_pt, rho_water);
+
+        _pressure_wetting[ip] = pg_int_pt - pc_int_pt;
+        // heat capacity of nonwet phase
+        double const heat_capacity_dry_air =
+            _process_data.material->getSpecificHeatCapacityAir(pg_int_pt,
+                                                               T_int_pt);
+        const double heat_capacity_water_vapor =
+            _process_data.material->getSpecificHeatCapacityVapor(pg_int_pt,
+                                                                 T_int_pt);
+
+        double const heat_capacity_water =
+            _process_data.material->getSpecificHeatCapacityWater(pg_int_pt,
+                                                                 T_int_pt);
+        double const heat_capacity_solid =
+            _process_data.material->getSpecificHeatCapacitySolid(pg_int_pt,
+                                                                 T_int_pt);
+
+        double const latent_heat_evaporation =
+            _process_data.latent_heat_evaporation(t, pos)[0];
+        double const enthalpy_nonwet_air =
+            heat_capacity_dry_air * (T_int_pt - CelsiusZeroInKelvin) +
```

Comment:
Why is the conversion K <-> Celsius needed? All temperatures, but for specific material models, should be in K.